sql/src/main/java/io/crate/metadata/Routing.java

```diff
@@ -128,6 +129,64 @@ public boolean containsShards(String nodeId) {
         return false;
     }
 
+    /**
+     * Update the locations of the current routing by merging them with the locations
+     * that have been computed in an existing routing for the same table.
+     * <p>
+     * if a shard has been already routed in {@paramref existingLocations}
+     * then: use the existing node
+     * else: use the new node
+     *
+     * @param existingLocations Existing locations routed for the same table
+     */
+    public void updateLocations(Map<String, Map<String, List<Integer>>> existingLocations) {
```

Comment:
wouldn't  `otherLocations` make more sense as argument. 
You're referring to `current` & `existing` which imho sounds like it could be the same thing anyway?